core/src/test/java/se/tre/freki/labels/LabelClientTypeContextTest.java

```diff
@@ -152,4 +155,67 @@ public void createIdPublishesEventOnSuccess() throws Exception {
     typeContext.createId("foo").get();
     verify(eventBus).post(any(LabelCreatedEvent.class));
   }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testRenameNewNameExists() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    // Can not rename to a name that is already taken
+    typeContext.createId("newName");
+    typeContext.rename("oldName", "newName");
+  }
+
+  @Test
+  public void testRenameIdNotFoundOnOldName() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    //No previous label called "oldName" return Failed future
+    ListenableFuture<Void> response = typeContext.rename("oldName", "newName");
+
+    Futures.addCallback(response, new FutureCallback<Void>() {
```

Comment:
This may not fail if anything in `#rename` or nested calls moves to a different thread. Have you tested? I'm not sure how JUnit handles this. A simple `#get` call on `response` and a subsequent `fail("Rename should have failed when it did not find the current name")` would work otherwise?